src/Compilers/Core/Portable/SourceGeneration/Nodes/InputNode.cs

```diff
@@ -35,7 +35,7 @@ public InputNode(Func<DriverStateTable.Builder, ImmutableArray<T>> getInput, IEq
         private InputNode(Func<DriverStateTable.Builder, ImmutableArray<T>> getInput, Action<IIncrementalGeneratorOutputNode>? registerOutput, IEqualityComparer<T>? inputComparer = null, IEqualityComparer<T>? comparer = null, string? name = null)
         {
             _getInput = getInput;
-            _comparer = comparer ?? EqualityComparer<T>.Default;
+            _comparer = comparer;
             _inputComparer = inputComparer ?? EqualityComparer<T>.Default;
```

Comment:
Should `_inputComparer` be handled similarly?

Reply:
No, we control the input comparer and all the inputs it compares. If that's throwing it's a real product bug and we don't want to falsely attribute it to a generator.